contrib/agent/build.gradle

```diff
@@ -155,21 +156,41 @@ checkstyleIntegrationTest.enabled = JavaVersion.current().isJava8Compatible()
 // Disable findbugs for integration tests, too.
 findbugsIntegrationTest.enabled = false
 
-// Run integration tests with the agent enabled.
-task integrationTest(type: Test) {
-  testLogging {
-    // Let Gradle output the stdout and stderr from tests, too. This is useful for investigating
-    // test failures on Travis, where we can't view Gradle's test reports.
-    showStandardStreams = true
+def javaExecutables = [ 'JAVA6_HOME', 'JAVA7_HOME', 'JAVA8_HOME' ]
+    .collect { System.getenv(it) }
+    .findAll { it != null }
+    .plus(System.getProperty('java.home'))
+    .collect { org.apache.tools.ant.taskdefs.condition.Os.isFamily(
+                   org.apache.tools.ant.taskdefs.condition.Os.FAMILY_WINDOWS)
+                   ? "${it}/bin/java.exe"
+                   : "${it}/bin/java" }
+    .findAll { new File(it).canExecute() }
+    .unique()
+
+assert javaExecutables.size > 0 :
+       'No Java executables found for running integration tests'
+
+task integrationTest
+
+javaExecutables.eachWithIndex { javaExecutable, index ->
```

Comment:
Is there a way to print that a specific JRE is in use? For example, right now, there's a matrix build in travis (which I guess amplifies this test scenario as you would test all JREs for both JDK 7 and 8). We might want a flag to only run the nested multi-version integration test with one of the matrixes and if so, print output so we know it is happening

Reply:
> Is there a way to print that a specific JRE is in use? For example, right now, there's a matrix build in travis (which I guess amplifies this test scenario as you would test all JREs for both JDK 7 and 8).

Yes, for debugging I had used something like 'doFirst { println "Testing with ${javaExecutable}..." }' and a similar 'doLast ...', but then removed it from this change. I'll add something like that (using the Gradle provided logger instead of plain printlns) - it might be just good enough to see what's going on.

> We might want a flag to only run the nested multi-version integration test with one of the matrixes and if so, print output so we know it is happening

Yes, this might be useful. I was thinking about using a single "java.homes" property (or environment variable?, TBD) for passing the list of (potential) java homes to Gradle instead of the separate environment variables JAVA6_HOME, JAVA7_HOME, and JAVA8_HOME. In the Travis build script, we could set this "java.homes" property differently for each build. I now like this idea better - let me know what you think before I make any changes in this direction.